scripts/pylib/pytest-twister-harness/src/twister_harness/fixtures/mcumgr.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import pytest
+import logging
+import re
+
+from typing import Generator
+from subprocess import check_output, getstatusoutput
+from pathlib import Path
+from dataclasses import dataclass
+
+from twister_harness.device.device_abstract import DeviceAbstract
+
+
+logger = logging.getLogger(__name__)
+
+
+class MCUmgrException(Exception):
+    """General MCUmgr exception."""
+
+
+@dataclass
+class MCUmgrImage:
+    image: int
+    slot: int
+    version: str = ''
+    flags: str = ''
+    hash: str = ''
+
+
+class MCUmgr:
+    """Sample wrapper for mcumgr command-line tool"""
+    mcumgr_exec = 'mcumgr'
+
+    def __init__(self, connection_options: str):
+        self.conn_opts = connection_options
+
+    @classmethod
+    def create_for_serial(cls, serial_port: str) -> MCUmgr:
+        return cls(connection_options=f'--conntype serial --connstring={serial_port}')
+
+    @classmethod
+    def is_available(cls) -> bool:
+        exitcode, output = getstatusoutput(f'{cls.mcumgr_exec} version')
+        if exitcode != 0:
+            logger.warning(f'mcumgr tool not available: {output}')
+            return False
+        return True
+
+    def run_command(self, cmd: str) -> str:
+        command = f'{self.mcumgr_exec} {self.conn_opts} {cmd}'
+        logger.info(f"CMD: {command}")
+        return check_output(command, shell=True, text=True)
```

Comment:
Please consider of using `check_output(shlex.split(command), text=True)` instead of `shell=True`, which is in general considered as potential source of vulnerabilities.
https://docs.python.org/3/library/subprocess.html#security-considerations
https://semgrep.dev/docs/cheat-sheets/python-command-injection/

Reply:
Thanks for comment. I agree and changed to use `shlex` to avoid `shell=True`